t/23-amqp-fedoraci.t

```diff
@@ -0,0 +1,242 @@
+    publish_amqp => sub {
+        my ($self, $topic, $data) = @_;
+        $published{$topic} = $data;
+    });
+
+my $schema = OpenQA::Test::Database->new->create();
+
+# this test also serves to test plugin loading via config file
+my @conf = (
+    "[global]\n",                                      "plugins=AMQP\n",
+    "base_url=https://openqa.stg.fedoraproject.org\n", "[amqp]\n",
+    "fedora_ci_messages=1\n"
+);
+my $tempdir = tempdir;
+$ENV{OPENQA_CONFIG} = $tempdir;
+path($ENV{OPENQA_CONFIG})->make_path->child("openqa.ini")->spurt(@conf);
+
+my $t = Test::Mojo->new('OpenQA::WebAPI');
+
+# XXX: Test::Mojo loses its app when setting a new ua
+# https://github.com/kraih/mojo/issues/598
+my $app = $t->app;
+$t->ua(
+    OpenQA::Client->new(apikey => 'PERCIVALKEY02', apisecret => 'PERCIVALSECRET02')->ioloop(Mojo::IOLoop->singleton));
+$t->app($app);
+
+my $settings = {
+    DISTRI      => 'Unicorn',
+    FLAVOR      => 'pink',
+    VERSION     => '42',
+    BUILD       => 'Fedora-Rawhide-20180129.n.0',
+    TEST        => 'rainbow',
+    ISO         => 'whatever.iso',
+    DESKTOP     => 'DESKTOP',
+    KVM         => 'KVM',
+    ISO_MAXSIZE => 1,
+    MACHINE     => "RainbowPC",
+    ARCH        => 'x86_64',
+    SUBVARIANT  => 'workstation'
+};
+
+my $expected_artifact = {
+    id           => 'Fedora-Rawhide-20180129.n.0',
+    iso          => 'whatever.iso',
+    type         => 'productmd-compose',
+    compose_type => 'nightly',
+};
+
+my $expected_contact = {
+    name  => 'Fedora openQA',
+    team  => 'Fedora QA',
+    url   => 'https://openqa.stg.fedoraproject.org',
+    docs  => 'https://fedoraproject.org/wiki/OpenQA',
+    irc   => '#fedora-qa',
+    email => 'qa-devel@lists.fedoraproject.org',
+};
+
+my $expected_pipeline = {
+    id   => 'openqa.Fedora-Rawhide-20180129.n.0.rainbow.RainbowPC.pink.x86_64',
+    name => 'openqa.Fedora-Rawhide-20180129.n.0.rainbow.RainbowPC.pink.x86_64',
+};
+
+my $expected_run = {
+    url => '',
+    log => '',
+    id  => '',
+};
+
+my $expected_system = {
+    os           => 'fedora-42',
+    provider     => 'openqa',
+    architecture => 'x86_64',
+    variant      => 'workstation',
+};
+
+my $expected_test = {
+    category  => 'validation',
+    type      => 'rainbow RainbowPC pink x86_64',
+    namespace => 'compose',
+    lifetime  => 240,
+};
+
+my $expected_error;
+
+my $expected_version = '0.2.1';
+
+sub get_expected {
+    my $expected = {
+        artifact => $expected_artifact,
+        contact  => $expected_contact,
+        pipeline => $expected_pipeline,
+        run      => $expected_run,
+        system   => $expected_system,
+        test     => $expected_test,
+        version  => $expected_version,
+    };
+    $expected->{'error'} = $expected_error if ($expected_error);
+    return $expected;
+}
+
+# create a job via API
+my $job;
+my $newjob;
+subtest 'create job' => sub {
+    my $post = $t->post_ok("/api/v1/jobs" => form => $settings)->status_is(200);
```

Comment:
The `$post` here is just an alias for `$t`, the variable is pointless.

Reply:
Same for the blocks below.

Reply:
Er. Are you sure? This is how it is in both the 18-fedmsg.t and 23-amqp.t tests, and I think I initially copied it from some other test for the fedmsg tests. My understanding is that `$t` at this point is a `Mojo::Test` instance, which is basically a test-modified `Mojo::UserAgent`. The [docs for `Mojo::UserAgent`](https://mojolicious.org/perldoc/Mojo/UserAgent#post) say that `post` (or `post_ok` in this case, for the test agent) returns an [instance of `Mojo::Transaction:HTTP`](https://mojolicious.org/perldoc/Mojo/Transaction/HTTP), which is clearly a different thing entirely, they don't say it returns itself. So I don't see how `$post` is an alias for `$t` in this case.

There's a couple of cases where we assign it and then just don't use it, those could be changed I guess.

Reply:
...huh, but then the code does seem like you have a point, because we use `$post->tx`. That's odd. Are the docs wrong or something?

Reply:
for the record: turns out this is a difference between the normal useragent and test::mojo, the normal useragent returns the transaction, methods in test::mojo like `post_ok` return the agent itself.